sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -3403,10 +3406,16 @@ def value_counts(self, subset=None, sort=False, normalize=False,
           "ordering on the dataset which likely will not be preserved.",
           reason="order-sensitive")
     columns = subset or list(self.columns)
-    result = self.groupby(columns).size()
+
+    if dropna:
+      dropped = self.dropna()
+    else:
+      dropped = self
+
+    result = dropped.groupby(columns, dropna=dropna).size()
 
     if normalize:
-      return result/self.dropna().length()
+      return result/dropped.length()
```

Comment:
This looks correct, but to be explicit this is changing the normalized value for dropna=False. Is this intentional?

Reply:
Yes, the pandas value_counts it defined for both dropna=False and True. I added a combinatorial test for dropna and normalize to verify this implementation and the pandas implementation.